### PR TITLE
subscribers -> futures

### LINKED ISSUE #215: migrate subscribers to futures
it's been discussed multiple times and it's gonna happen. subscribers cannot be chained and stuff can get messy quick.

### PR DESCRIPTION
**Pull request by MrIvanPlays**
Closes #215

@Jikoo and @lokka30 you can do work in that branch too.

thoughts wanted

### CODE REVIEW

**Comment by MrIvanPlays** [comment]
@Jikoo please when you have time fix the migration code. thanks in advance

File: platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/economy/BalanceCache.java [hunk 1]
```diff
@@ -99,58 +101,65 @@ private void proceed(
...
+            if (ex != null) {
+                throw new RuntimeException("An error occurred whilst updating balance cache", ex);
+            }
+
+            if (!res.isSuccessful()) {
+                // log the problem and proceed with next entry
+                TreasuryPlugin.getInstance().logger().error(
+                        "Error whilst trying to update balance cache for ( " + (player.getName() != null
+                                ? player.getName()
+                                : player.getUniqueId().toString()) + "): " + res
```
Comment (lokka30) [inline platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/economy/BalanceCache.java:119]:
```suggestion
                        "Error whilst trying to update balance cache for " + (player.getName() != null
                                ? player.getName()
                                : player.getUniqueId().toString()) + ": " + res
```

**Review by lokka30 — APPROVED** [review]

**Comment by lokka30** [comment]
We may want to squash some commits before we commit to master. 😁

**Comment by lokka30** [comment]
🎉🎉
It has finally come. Thanks Ivan!

**Pull request merged**